From 0005440acd5e979f75584ec7f7244c9d554a416e Mon Sep 17 00:00:00 2001
From: Rajesh Bhagat <rajesh.bhagat@nxp.com>
Date: Thu, 8 Feb 2018 02:32:27 +0530
Subject: [PATCH 11/31] board: freescale: ls1043adcm: Move CONFIG_ENV_IS_IN*
 from deconfig files

Move CONFIG_ENV_IS_IN* from deconfig files to ls1043adcm.h

Signed-off-by: Rajesh Bhagat <rajesh.bhagat@nxp.com>
---
 configs/ls1043adcm_defconfig        | 1 -
 configs/ls1043adcm_nand_defconfig   | 1 -
 configs/ls1043adcm_sdcard_defconfig | 1 -
 include/configs/ls1043adcm.h        | 5 +++++
 4 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/configs/ls1043adcm_defconfig b/configs/ls1043adcm_defconfig
index cc93553315..2bf109643d 100644
--- a/configs/ls1043adcm_defconfig
+++ b/configs/ls1043adcm_defconfig
@@ -18,7 +18,6 @@ CONFIG_CMD_USB=y
 CONFIG_CMD_CACHE=y
 CONFIG_MTDPARTS_DEFAULT="mtdparts=60000000.nor:2m@0x100000(nor_bank0_uboot),40m@0x1100000(nor_bank0_fit),7m(nor_bank0_user),2m@0x4100000(nor_bank4_uboot),40m@0x5100000(nor_bank4_fit),-(nor_bank4_user);7e800000.flash:1m(nand_uboot),1m(nand_uboot_env),20m(nand_fit);spi0.0:1m(uboot),5m(kernel),1m(dtb),9m(file_system)"
 CONFIG_OF_CONTROL=y
-CONFIG_ENV_IS_IN_SPI_FLASH=y
 CONFIG_DM=y
 CONFIG_FSL_CAAM=y
 CONFIG_SPI_FLASH=y
diff --git a/configs/ls1043adcm_nand_defconfig b/configs/ls1043adcm_nand_defconfig
index a8733b3344..b281a97028 100644
--- a/configs/ls1043adcm_nand_defconfig
+++ b/configs/ls1043adcm_nand_defconfig
@@ -40,7 +40,6 @@ CONFIG_CMD_FAT=y
 CONFIG_MTDPARTS_DEFAULT="mtdparts=60000000.nor:2m@0x100000(nor_bank0_uboot),40m@0x1100000(nor_bank0_fit),7m(nor_bank0_user),2m@0x4100000(nor_bank4_uboot),40m@0x5100000(nor_bank4_fit),-(nor_bank4_user);7e800000.flash:1m(nand_uboot),1m(nand_uboot_env),20m(nand_fit);spi0.0:1m(uboot),5m(kernel),1m(dtb),9m(file_system)"
 # CONFIG_SPL_EFI_PARTITION is not set
 CONFIG_OF_CONTROL=y
-CONFIG_ENV_IS_IN_NAND=y
 CONFIG_DM=y
 CONFIG_FSL_CAAM=y
 CONFIG_MTD_NOR_FLASH=y
diff --git a/configs/ls1043adcm_sdcard_defconfig b/configs/ls1043adcm_sdcard_defconfig
index 5b1b68be01..e603184f3c 100644
--- a/configs/ls1043adcm_sdcard_defconfig
+++ b/configs/ls1043adcm_sdcard_defconfig
@@ -34,7 +34,6 @@ CONFIG_CMD_CACHE=y
 CONFIG_MTDPARTS_DEFAULT="mtdparts=60000000.nor:2m@0x100000(nor_bank0_uboot),40m@0x1100000(nor_bank0_fit),7m(nor_bank0_user),2m@0x4100000(nor_bank4_uboot),40m@0x5100000(nor_bank4_fit),-(nor_bank4_user);7e800000.flash:1m(nand_uboot),1m(nand_uboot_env),20m(nand_fit);spi0.0:1m(uboot),5m(kernel),1m(dtb),9m(file_system)"
 # CONFIG_SPL_EFI_PARTITION is not set
 CONFIG_OF_CONTROL=y
-CONFIG_ENV_IS_IN_MMC=y
 CONFIG_DM=y
 CONFIG_FSL_CAAM=y
 CONFIG_MTD_NOR_FLASH=y
diff --git a/include/configs/ls1043adcm.h b/include/configs/ls1043adcm.h
index e70a18bc41..a0e8f92fa7 100644
--- a/include/configs/ls1043adcm.h
+++ b/include/configs/ls1043adcm.h
@@ -247,13 +247,18 @@
 #endif
 
 #if defined(CONFIG_NAND_BOOT)
+#define CONFIG_ENV_IS_IN_NAND
 #define CONFIG_ENV_SIZE			0x2000
 #define CONFIG_ENV_OFFSET		(24 * CONFIG_SYS_NAND_BLOCK_SIZE)
 #elif defined(CONFIG_SD_BOOT)
+#define CONFIG_ENV_IS_IN_MMC
 #define CONFIG_ENV_OFFSET		(3 * 1024 * 1024)
 #define CONFIG_SYS_MMC_ENV_DEV		0
 #define CONFIG_ENV_SIZE			0x2000
 #elif defined(CONFIG_QSPI_BOOT)
+#define SYS_NO_FLASH
+#undef CONFIG_CMD_IMLS
+#define CONFIG_ENV_IS_IN_SPI_FLASH
 #define CONFIG_ENV_SIZE			0x2000		/* 8KB */
 #define CONFIG_ENV_OFFSET		0x300000	/* 3MB */
 #define CONFIG_ENV_SECT_SIZE		0x40000		/* 256KB */
-- 
2.17.1

